tests/unit/core/utils/test_index.js

```diff
@@ -761,4 +764,32 @@ describe(__filename, () => {
       expect(isTheme(ADDON_TYPE_EXTENSION)).toEqual(false);
     });
   });
+
+  describe('addQueryParamsToHistory', () => {
+    it('adds a query object to history.location', () => {
+      const history = createFakeHistory({
+        location: createFakeLocation({ query: null }),
+      });
+
+      expect(history).toHaveProperty('location.query', null);
+
+      const historyWithQueryParams = addQueryParamsToHistory({ history });
+
+      expect(historyWithQueryParams).toHaveProperty('location.query', {});
+    });
+
+    it('parses the query string to build the query object', () => {
+      const history = createFakeHistory({
+        location: createFakeLocation({ query: null, search: 'foo=123' }),
+      });
+
+      expect(history).toHaveProperty('location.query', null);
+
+      const historyWithQueryParams = addQueryParamsToHistory({ history });
+
+      expect(historyWithQueryParams.location.query).toEqual({
```

Comment:
In the test above you use `expect(historyWithQueryParams).toHaveProperty('location.query', ...)` can you not use that here as well?

Reply:
good catch, I can do it!